### PR TITLE
[CS-742] Don't cache redirects, cache-control on s3 files for cloudfront.

### PR DESCRIPTION
**Pull request by gb-dsigmeth**
Redirect to the correct url
Don't cache the redirect
Return a 4 hour cache header for resized images

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
-  const width = parseInt(match[2], 10);
-  const height = parseInt(match[3], 10);
+  const width = match[2] === 'auto' ? null : parseInt(match[2], 10);
+  const height = match[3] === 'auto' ? null : parseInt(match[3], 10);
   const originalKey = match[4];
 
-  if(ALLOWED_DIMENSIONS.size > 0 && !ALLOWED_DIMENSIONS.has(dimensions)) {
-     callback(null, {
+  if (ALLOWED_DIMENSIONS.size > 0 && !ALLOWED_DIMENSIONS.has(dimensions)) {
+    callback(null, {
```
Comment (reganmeloche) [inline index.js]: It's not part of this PR, but just curious why this status is a 403

File: index.js [hunk 2]
```diff
@@ -1,55 +1,61 @@
...
-        Body: buffer,
-        Bucket: BUCKET,
-        ContentType: 'image/png',
-        Key: key,
-      }).promise()
+      Body: buffer,
+      Bucket: BUCKET,
+      ContentType: 'image/png',
+      Key: key,
+      CacheControl: 'max-age=14400', // 4 hours
```
Comment (reganmeloche) [inline index.js]: It might be useful to have this max-age defined as a constant somewhere?

File: index.js (hunk 1 above)
Comment (gb-dsigmeth) [reply index.js]: 403 is Forbidden, So I think the idea is that the provided size is not allowed, and is thus forbidden, but we've pretty consistently used 400s in the case of bad parameters, so I'm fine to change it to a 400.

**Review by normrider2001 — APPROVED** [review]
